controllers/suite_test.go

```diff
@@ -270,95 +263,76 @@ func newParams(taContainerImage string, file string) (manifests.Params, error) {
-					Image:   taContainerImage,
+	return v1alpha1.OpenTelemetryCollector{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "opentelemetry.io",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "default",
+		},
+		Spec: v1alpha1.OpenTelemetryCollectorSpec{
+			Mode: v1alpha1.ModeStatefulSet,
+			Ports: []v1.ServicePort{{
+				Name: "web",
+				Port: 80,
+				TargetPort: intstr.IntOrString{
+					Type:   intstr.Int,
+					IntVal: 80,
 				},
-				Replicas: &replicas,
-				Config:   string(configYAML),
+				NodePort: 0,
+			}},
+			TargetAllocator: v1alpha1.OpenTelemetryTargetAllocator{
+				Enabled: true,
+				Image:   taContainerImage,
 			},
+			Replicas: &replicas,
+			Config:   string(configYAML),
 		},
-		Scheme: testScheme,
-		Log:    logger,
 	}, nil
 }
 
-func paramsWithHPA(minReps, maxReps int32) manifests.Params {
+func paramsWithHPA(minReps, maxReps int32) v1alpha1.OpenTelemetryCollector {
 	configYAML, err := os.ReadFile("testdata/test.yaml")
 	if err != nil {
 		fmt.Printf("Error getting yaml file: %v", err)
 	}
-
 	cpuUtilization := int32(90)
 
-	configuration := config.New(config.WithCollectorImage(defaultCollectorImage), config.WithTargetAllocatorImage(defaultTaAllocationImage))
-
-	return manifests.Params{
-		Config: configuration,
-		Client: k8sClient,
-		OtelCol: v1alpha1.OpenTelemetryCollector{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       "opentelemetry.io",
-				APIVersion: "v1",
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "hpatest",
-				Namespace: "default",
-				UID:       instanceUID,
-			},
-			Spec: v1alpha1.OpenTelemetryCollectorSpec{
-				Ports: []v1.ServicePort{{
-					Name: "web",
-					Port: 80,
-					TargetPort: intstr.IntOrString{
-						Type:   intstr.Int,
-						IntVal: 80,
-					},
-					NodePort: 0,
-				}},
-				Config: string(configYAML),
-				Autoscaler: &v1alpha1.AutoscalerSpec{
-					MinReplicas:          &minReps,
-					MaxReplicas:          &maxReps,
-					TargetCPUUtilization: &cpuUtilization,
+	return v1alpha1.OpenTelemetryCollector{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "opentelemetry.io",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "hpatest",
+			Namespace: "default",
+			UID:       instanceUID,
+		},
+		Spec: v1alpha1.OpenTelemetryCollectorSpec{
+			Ports: []v1.ServicePort{{
+				Name: "web",
+				Port: 80,
+				TargetPort: intstr.IntOrString{
+					Type:   intstr.Int,
+					IntVal: 80,
 				},
+				NodePort: 0,
+			}},
+			Config: string(configYAML),
+			Autoscaler: &v1alpha1.AutoscalerSpec{
+				MinReplicas:          &minReps,
+				MaxReplicas:          &maxReps,
+				TargetCPUUtilization: &cpuUtilization,
 			},
 		},
-		Scheme:   testScheme,
-		Log:      logger,
-		Recorder: record.NewFakeRecorder(10),
 	}
 }
 
-func paramsWithPolicy(minAvailable, maxUnavailable int32) manifests.Params {
+func paramsWithPolicy(minAvailable, maxUnavailable int32) v1alpha1.OpenTelemetryCollector {
```

Comment:
We should rename this function if we're changing the return type.

Reply:
agreed, will make the change :)